### PR TITLE
User Cannot Post Duplicate Favorites

### LINKED ISSUE #13: User cannot have duplicate favorites
- Duplicates are defined as having the same title and artistName

When a user sends a request POST /api/v1/favorites with a body including a title and artistName and there and already a favorite with that title and artistName in the database, they receive a response with a 409 and a message that a duplicate favorite cannot be created.

Example request:
```
POST  /api/v1/favorites
{ title: "We Will Rock You", artistName: "Queen" }
```

Example response:
```
status 409
body:
{
  "error": "Unable to create favorite."
  "detail": "That song is already favorited."
}
```

### PR DESCRIPTION
**Pull request by rhantak**
- Add check in POST `/favorites` to see if request would cause a duplicate record
- Send a 409 error and refuse to create the favorite if this is the case

Closes #13

### CODE REVIEW

File: tests/favorites.spec.js [hunk 1]
```diff
@@ -227,5 +227,58 @@ describe('Test the favorites route', () => {
...
+        .send(newFav);
+
+      const res_2 = await request(app)
+        .post("/api/v1/favorites")
+        .send(newFav);
+
+      let db_favorites = await database('favorites').select()
+        .then(result => {
+          return result;
+        })
```
Comment (ap2322) [inline tests/favorites.spec.js]: This definitely works. A more explicit test might try to find all instances in the db with the potentially duplicate information and make sure that length is 1. Since we're resetting the test db in between, this works for me too.

File: routes/api/v1/favorites.js [hunk 2]
```diff
@@ -38,12 +38,21 @@ router.post('/', (request, response) => {
...
+        database('favorites').where({'title': newFavorite.title, 'artistName': newFavorite.artistName}).select()
+          .then(faves => {
+            if (!faves.length) {
+              database('favorites')
+              .insert(newFavorite, ["id", "title", "artistName", "genre", "rating"])
+              .then(favoriteInfo =>{
+                response.status(201).send(favoriteInfo[0])
+              })
+            } else {
+              response.status(409).json({
```
Comment (ap2322) [inline routes/api/v1/favorites.js]: Well deliberated

**Comment by rhantak** [comment]
Thanks for refactoring that test! Looks quite a bit less messy now.

File: tests/favorites.spec.js (hunk 1 above)
Comment (rhantak) [reply tests/favorites.spec.js]: Yeah and I think that if we had multiple different records that would be the way to go. As is this should work unless something changes in our tests.
